### ISSUE TITLE
Adding Kobo Touch N905 (kobo-n905-imx507)

### ISSUE THREAD

**Issue by xy2iii** [postmarketOS/pmbootstrap#1047]
I'm opening this as a stub issue until a PR gets going.

**There are two motherboards revisions of the N905.** The main difference between the two is a different processor, as well as the presence of an SD card to replace eMMC. I am speaking on the later model, the one with an SD card slot, an imx507 (read the processor notes for this), and gold-ish contours in the middle of the motherboard.. As all data is stored on an SD, this makes it very easy to debug even what would be a brick on a normal phone (wrong flashing/does not boot). 

# Why?
This device can be bought for about 35$ online, it's very cheap, has open-sourced kernel, no binary blobs needed for core functionality (WiFi is blobbed, not much else) and is e-ink, which means it can last well weeks powered on because the e-ink screen uses no battery. It is so good that [people actually use the Kobo as a flight computer!](https://www.youtube.com/watch?v=IX_mmPqukqQ) (There's an entire forum dedicated to this.)

# Kernel, u-boot, etc sources
https://github.com/kobolabs/Kobo-Reader/tree/master/hw/imx508

# Interesting notes
The Kobo software (which I'll refer as 'nickel' from now on) is self-contained within the internal SD card. In theory, this means that you can keep nickel on the internal SD card while installing pmOS on the external one, and choosing at boot via u-boot which to choose. In practice, [this has been done](https://sites.google.com/site/gibekm/hardware/kobo/kobo-as-tablet). Interestingly, the functionality to boot from an SD card is already included in the native u-boot, all we need is a boot option to trigger it. [This work has already been done and can be modified easily if needed.](https://sites.google.com/site/gibekm/hardware/kobo/kobo-as-tablet)

**Comment by ollieparanoid** [comment]
Oh yeah, we need postmarketOS on an e-ink device \o/

Awesome research you did there! If you don't mind, it would be very helpful if you could also create a [device page](https://wiki.postmarketos.org/wiki/Help:Device_Page) on the wiki, so it's easier for people with the same device to see that you're working on this already. You could link to this issue from the wiki.

Just in case this device runs a legacy version of u-boot, which requires the legacy u-boot initfs format: When you set `deviceinfo_generate_legacy_uboot_initfs="true"`, then it will automatically get generated (see `device-nokia-rx51`).

Let us know if you're stuck somewhere!
Looking forward for more work and a photo once it's ready 😃

**Comment by im-grey** [comment]
This may be a stupid question, but would any of this apply to getting PMOS to run on a Barns & Noble Nook Simple Touch? I'm dying for it to happen.

**Comment by ata2001** [comment]
@im-grey not really, Nook Simple Touch has an OMAP3 SoC. Nevertheless, it should be possible to port pmos to that device too. You could try porting it yourself if you're interested.
